controllers/storagecluster/provider_server.go

```diff
@@ -147,8 +147,8 @@ func (o *ocsProviderServer) createDeployment(r *StorageClusterReconciler, instan
 func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
 
 	if instance.Spec.ProviderAPIServerServiceType != "" && instance.Spec.ProviderAPIServerServiceType != corev1.ServiceTypeNodePort &&
-		instance.Spec.ProviderAPIServerServiceType != corev1.ServiceTypeLoadBalancer {
-		err := fmt.Errorf("providerAPIServer only supports service of type %s and %s", corev1.ServiceTypeNodePort, corev1.ServiceTypeLoadBalancer)
+		instance.Spec.ProviderAPIServerServiceType != corev1.ServiceTypeLoadBalancer && instance.Spec.ProviderAPIServerServiceType != corev1.ServiceTypeClusterIP {
```

Comment:
- this line is getting busy, maybe simplify w/ `switch/case` 🤔 
``` go
if instance.Spec.ProviderAPIServerServiceType != "" {
  switch instance.Spec.ProviderAPIServerServiceType {
    case corev1.ServiceTypeClusterIP, corev1.ServiceTypeNodePort, corev1.ServiceTypeLoadBalancer:
    default:
      // log and return error
  }
}

Reply:
Done